tests/conformance/FirestoreTest.php

```diff
@@ -0,0 +1,368 @@
+<?php
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Tests\Conformance;
+
+use Google\Cloud\Core\Timestamp;
+use Google\Cloud\Firestore\Connection\ConnectionInterface;
+use Google\Cloud\Firestore\FieldPath;
+use Google\Cloud\Firestore\FieldValue;
+use Google\Cloud\Firestore\FirestoreClient;
+use Google\Cloud\Firestore\PathTrait;
+use Google\Cloud\Tests\ArrayHasSameValuesToken;
+use Google\GAX\Serializer;
+use GuzzleHttp\Client;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+use Prophecy\Exception\Call\UnexpectedCallException;
+
+/**
+ * @group firestore
+ */
+class FirestoreTest extends TestCase
+{
+    use PathTrait;
+
+    const TEST_FILE = 'https://raw.githubusercontent.com/GoogleCloudPlatform/google-cloud-common/master/testing/firestore/testdata/tests.binprotos';
+
+    private $client;
+    private $connection;
+
+    private $skipped = [
```

Comment:
Some tests are skipped due to a particular feature not being supported yet.

1. `need to strip empty maps`. When a server timestamp sentinel is nested within a map, the value is removed as expected, but the empty map value is preserved in the request. This doesn't cause any issue, because the map is immediately populated with the server timestamp via the `transform` write. However, the test expects that the empty map is stripped from the request. This will be addressed with additional improvements to the ValueMapper class.
* `need mergeFields support`. The PHP client doesn't have support for `(array) $options.mergeFields` on `set()` at this time. `(bool) $options.merge` is supported and tested. I'd like to add this post-release, but need @schmidt-sebastian's input.

Reply:
Both of these things are fine as is. 

(1) should be fixed to make the conformance tests pass (but you can treat this as a P2 as it does not affect the usability of the client.

(2) is optional and can be added after the release.

Thanks so much for porting these tests!